hcipy/wavefront_sensing/pyramid.py

```diff
@@ -17,18 +18,43 @@ class ModulatedPyramidWavefrontSensorOptics(WavefrontSensorOptics):
 		The modulation radius in radians.
 	num_steps : int
 		The number of steps per modulation cycle.
+	fast_modulation_method : boolean
+		If True the fast propagation method will be used. Default is False.
 	'''
-	def __init__(self, pyramid_wavefront_sensor, modulation, num_steps=12):
+	def __init__(self, pyramid_wavefront_sensor, modulation, num_steps=12, fast_modulation_method=False):
 		self.modulation = modulation
 		self.pyramid_wavefront_sensor = pyramid_wavefront_sensor
 		self.tip_tilt_mirror = TipTiltMirror(self.pyramid_wavefront_sensor.input_grid)
+		self.focal_grid = self.pyramid_wavefront_sensor.focal_grid
 
+		# Instead of shifting the input beam, we shift the pyramid prism.
 		theta = np.linspace(0, 2 * np.pi, num_steps, endpoint=False)
 		x_modulation = modulation / 2 * np.cos(theta)
 		y_modulation = modulation / 2 * np.sin(theta)
 
 		self.modulation_positions = CartesianGrid(UnstructuredCoords((x_modulation, y_modulation)))
 
+		separation = self.pyramid_wavefront_sensor._separation
+		refractive_index = self.pyramid_wavefront_sensor._refractive_index
+		wavelength_0 = self.pyramid_wavefront_sensor._wavelength_0
+
+		def surface_function(temp_grid):
+			return Field(-separation / (2 * (refractive_index(wavelength_0) - 1)) * (np.abs(temp_grid.x) + np.abs(temp_grid.y)), temp_grid)
+
+		self._pyramid_apodizers = []
+		for p in self.modulation_positions:
+			shifted_grid = self.focal_grid.shifted(2 * p)  # This factor two is to compensate for the 1/2 for the mirror surface.
```

Comment:
I prefer to have this inline comment on a separate line. Use inline comments sparingly.

Reply:
done